source/js/apiHelpers.js

```diff
@@ -40,24 +40,30 @@ async function getDetailedRecipeInfoBulk(ids) {
  */
 function extractIDs(data) {
   const { results } = data;
+  if (!results) {
+    return [];
+  }
   const ids = [];
   results.forEach((result) => {
+    // TODO: figure out how to check if current id is already in local storage
     ids.push(result.id);
   });
   return ids;
 }
 
 /**
  * Get recipes by keywords(user searching for recipes)
- * @param {Number} num - max number of recipes to get
  * @param {String} query - Keywords to search for
+ * @param {Number} num - max number of recipes to get
+ * @param {Number} [offset=0] - number of recipes to skip
+ *  (use random number so we dont get same results everytime)
  * @returns {Object} list of recipes with detailed info
  */
 // eslint-disable-next-line no-unused-vars
-async function getRecipesByName(query, num) {
+async function getRecipesByName(query, num, offset = 0) {
   return new Promise((resolve, reject) => {
     const queryFormatted = query.trim().replace(/\s+/g, '-').toLowerCase();
-    fetch(`https://${HOST}/recipes/complexSearch?&query=${queryFormatted}&number=${num}&sort=populatrity`, {
+    fetch(`https://${HOST}/recipes/complexSearch?&query=${queryFormatted}&number=${num}&sort=populatrity&offset=${offset}`, {
```

Comment:
kinda unrelated but does this still work if parameters are missing, like if we just pass query and no number? might be good to put a default value there too but doesnt matter if its working. also there's a typo in popularity

Reply:
yea I can add default values